serializers/src/glTF/2.0/babylon.glTFAnimation.ts

```diff
@@ -0,0 +1,718 @@
+            let nextKeyFrame: Nullable<IAnimationKey> = null;
+            let prevKeyFrame: Nullable<IAnimationKey> = null;
+            let endFrame: Nullable<number> = null;
+            minMaxFrames.min = this.fround(minFrame / fps);
+
+            if (frameDelta / fps > sampleRate) {
+                let keyFrames = animation.getKeys();
+                for (let i = 0, length = keyFrames.length; i < length; ++i) {
+                    endFrame = null;
+                    currKeyFrame = keyFrames[i];
+
+                    if (i + 1 < length) {
+                        nextKeyFrame = keyFrames[i + 1];
+                        if (_GLTFAnimation._ValuesAreEqual(currKeyFrame.value, nextKeyFrame.value)) {
+                            if (i === 0) { // set the first frame to itself
+                                endFrame = currKeyFrame.frame;
+                            }
+                            else {
+                                continue;
+                            }
+                        }
+                        else {
+                            endFrame = nextKeyFrame.frame;
+                        }
+                    }
+                    else {
+                        // at the last key frame
+                        prevKeyFrame = keyFrames[i - 1];
+                        if (_GLTFAnimation._ValuesAreEqual(currKeyFrame.value, prevKeyFrame.value)) {
+                            continue;
+                        }
+                        else {
+                            endFrame = maxFrame;
+                        }
+                    }
+                    if (endFrame) {
+                        for (let f = currKeyFrame.frame; f <= endFrame; f += sampleRate) {
+                            time = this.fround(f / fps);
+                            if (time === previousTime) {
+                                continue;
+                            }
+                            previousTime = time;
+                            maxUsedFrame = time;
+                            value = animation._interpolate(f, 0, undefined, animation.loopMode);
+                            _GLTFAnimation._SetInterpolatedValue(value, time, animationChannelTargetPath, quaternionCache, inputs, outputs, convertToRightHandedSystem, useQuaternion);
+                        }
+                    }
+                }
+                if (maxUsedFrame) {
+                    minMaxFrames.max = maxUsedFrame;
+                }
+            }
+            else {
+                Tools.Warn('Frame range is too small to bake animation');
+            }
+        }
+
+        /**
+         * Checks if two values are equal to each other
+         * @param value1 First value as a Vector3 or Quaternion
+         * @param value2 Second value as a Vector3 or Quaternion
+         */
+        private static _ValuesAreEqual(value1: Vector3 | Quaternion, value2: Vector3 | Quaternion): boolean {

```

Comment:
not sure this is necessary. typescript should see that both types have an equal function